Backend/Tournament_Public/controllers/publicUserController.js

```diff
@@ -0,0 +1,151 @@
+const TournamentUserService = require('../userService/userService');
+
+
+
+/** Controller to get all tournaments
+ * @async
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Json} Return Upcoming tournament based on upcoming dates.
+ * @throws {error} If input invalid or missing fields.
+ */
+
+exports.getAllTournaments = async (req, res, next) => {
+    try {
+        const allTournaments = await TournamentUserService.getAllTournaments();
+        if (!allTournaments || allTournaments.length === 0) {
+            const error = new Error('No tournaments retrieved');
```

Comment:
Think can maybe change the comment to 'No tournaments found' instead or smt I'm assuming only 0 if got no tournaments ah